File: 2626-count-the-number-of-good-subarrays/2626-count-the-number-of-good-subarrays.java
class Solution {
    public long countGood(int[] nums, int k) {
        long ans = 0, pairs = 0;
        Map<Integer,Integer> mp = new HashMap<>();
        int n = nums.length;
        int i = 0, j = 0;

        while(j < n){
            pairs += mp.getOrDefault(nums[j],0);
            mp.put(nums[j],mp.getOrDefault(nums[j],0)+1);

            while(pairs >= k){
                ans += (n-j);
                mp.put(nums[i], mp.get(nums[i])-1);
                pairs -= mp.get(nums[i]);
                i++;
            }
            j++;
        }
        return ans;
    }
}

File: 2626-count-the-number-of-good-subarrays/2626-count-the-number-of-good-subarrays.cpp
class Solution {
public:
    long long countGood(vector<int>& nums, int k) {
        long long pairs = 0;
        long long ans = 0;
        unordered_map<int,int> mp;
        int n = nums.size();
        int i = 0, j = 0;
        while(j < n){
            pairs += mp[nums[j]];
            mp[nums[j]]++;
            while(pairs >= k){
                ans += (n-j);
                mp[nums[i]]--;
                pairs -= mp[nums[i]];
                i++;
            }
            j++;
        }
        return ans;
    }
};